chebi-update/src/main/java/org/reactome/release/chebiupdate/ChebiUpdater.java

```diff
@@ -383,9 +496,36 @@ public void checkForDuplicates() throws SQLException, Exception
 	 * @param refMolecules - a list of ReferenceMolecules. The Identifier of each of these molecules will be sent to ChEBI to get up-to-date information for that Identifier.
 	 * @param entityMap - a ReferenceMolecule DB_ID-to-ChEBI Entity map. Will be updated by this method.
 	 * @param failedEntitiesList - A list of ReferenceMolecules for which no information was returned by ChEBI. Will be updated by this method.
+	 * @throws IOException 
 	 */
-	private void retrieveUpdatesFromChebi(Collection<GKInstance> refMolecules, Map<Long, Entity> entityMap, List<GKInstance> failedEntitiesList)
+	private void retrieveUpdatesFromChebi(Collection<GKInstance> refMolecules, Map<Long, Entity> entityMap, Map<GKInstance, String> failedEntitiesList) throws IOException
 	{
+		Map<String,List<String>> chebiCache = Collections.synchronizedMap(new HashMap<String, List<String>>());
+		if (this.useCache)
+		{
+			logger.info("useCache is TRUE - chebi-cache file will be read, and populated. Identifiers not in the cache will be queried from ChEBI.");
+			// if the cache exists, load it.
+			if (Files.exists(Paths.get("chebi-cache")))
+			{
+				Files.readAllLines(Paths.get("chebi-cache")).parallelStream().forEach( line -> {
+					String[] parts = line.split("\t");
+					String oldChebiID = parts[0];
+					String newChebiID = parts[1];
+					String name = parts[2];
+					String formula = parts.length > 2 ? parts[3] : "";
+					chebiCache.put(oldChebiID, Arrays.asList(newChebiID, name, formula));
+				});
+			}
+			logger.debug("{} entries in the chebi-cache", chebiCache.size());
+		}
+		else
+		{
+			logger.info("useCache is FALSE - chebi-cache will NOT be read. ChEBI will be queried for ALL identifiers.");
+		}
```

Comment:
This could be extracted to a method to populate the chebiCache

Reply:
It could be, but I am not sure much value would be gained from that in this case.

Reply:
It would just shorten the method a bit.  I found it a bit difficult to keep in my head what was going on for over 100 lines.